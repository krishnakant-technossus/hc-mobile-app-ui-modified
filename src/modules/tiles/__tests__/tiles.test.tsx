// npx jest ./src/modules/tiles/__tests__/tiles.test.ts

import * as React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import {Text} from 'react-native';

import Tiles from '../tiles';
import ThemeProvider from '../../theming/theme-provider';
import defaultTheme from '../../theming/default-theme';

let items;

describe('Tiles', () => {
  beforeEach(() => {
    items = [
      {
        disabled: true,
        onPress: jest.fn(),
        id: 'tile-group-tile-worship',
        tileColor: 'primaryLight',
        title: 'Worship',
      },
      {
        onPress: jest.fn(),
        id: 'tile-group-tile-baptism',
        tileContent: <Text>X</Text>,
        tileColor: 'secondaryLight',
        title: 'Baptism',
      },
      {
        onPress: jest.fn(),
        id: 'tile-group-tile-prayer1',
        tileContent: <Text>X</Text>,
        tileColor: 'successLight',
        title: 'Prayer',
      },
    ];
  });

  describe('Rendering', () => {
    it('renders itself correctly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <Tiles items={items} />
        </ThemeProvider>,
      );

      const tiles = wrapper.getByTestId('tiles');

      expect(tiles.children).toHaveLength(items.length);

      expect(tiles).toBeDefined();

      expect(wrapper).toMatchSnapshot();
    });

    it('renders the children correctly', () => {
      const {getByText, queryAllByTestId} = render(
        <ThemeProvider theme={defaultTheme}>
          <Tiles items={items} />
        </ThemeProvider>,
      );

      const tileItems = queryAllByTestId('tile');

      expect(tileItems).toHaveLength(items.length);

      const option1 = getByText(items[0].title);
      const option2 = getByText(items[1].title);
      const option3 = getByText(items[2].title);

      expect(option1).toBeDefined();
      expect(option2).toBeDefined();
      expect(option3).toBeDefined();
    });
  });

  describe('Interactions', () => {
    it('should call item.onPress function when pressed', () => {
      const {queryAllByTestId} = render(
        <ThemeProvider theme={defaultTheme}>
          <Tiles items={items} />
        </ThemeProvider>,
      );

      const tileItems = queryAllByTestId('tile');

      expect(tileItems).toHaveLength(items.length);

      const [, option2] = tileItems;

      fireEvent.press(option2);

      expect(items[1].onPress).toHaveBeenCalledTimes(1);
    });

    it('should not call item.onPress function when pressed when item.disabled is true', () => {
      const {queryAllByTestId} = render(
        <ThemeProvider theme={defaultTheme}>
          <Tiles items={items} />
        </ThemeProvider>,
      );

      const tileItems = queryAllByTestId('tile');

      expect(tileItems).toHaveLength(items.length);

      const [option1] = tileItems;

      fireEvent.press(option1);

      expect(items[0].onPress).not.toHaveBeenCalled();
    });
  });
});
