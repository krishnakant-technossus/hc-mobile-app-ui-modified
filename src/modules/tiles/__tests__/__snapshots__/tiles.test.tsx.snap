// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tiles Rendering renders itself correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "width": "100%",
      },
    ]
  }
  testID="tiles"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onLayout={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": -6,
        "opacity": 0.25,
        "overflow": "hidden",
        "paddingBottom": 6,
        "paddingLeft": 6,
        "paddingRight": 6,
        "paddingTop": 6,
        "width": "50%",
      }
    }
    testID="tile"
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#1C2530",
            "borderRadius": 5,
            "height": "100%",
            "justifyContent": "space-between",
            "paddingBottom": 11,
            "paddingLeft": 11,
            "paddingRight": 11,
            "paddingTop": 11,
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignSelf": "flex-end",
              "marginBottom": 0,
            },
          ]
        }
        testID="tile-content"
      />
      <Text
        inversed={true}
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "bottom": 8,
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "700",
              "left": 6,
              "marginLeft": 4,
              "position": "absolute",
              "textAlign": "left",
            },
          ]
        }
        testID="tile-title"
        weight="bold"
      >
        Worship
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onLayout={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": -6,
        "opacity": 1,
        "overflow": "hidden",
        "paddingBottom": 6,
        "paddingLeft": 6,
        "paddingRight": 6,
        "paddingTop": 6,
        "width": "50%",
      }
    }
    testID="tile"
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#1C2530",
            "borderRadius": 5,
            "height": "100%",
            "justifyContent": "space-between",
            "paddingBottom": 11,
            "paddingLeft": 11,
            "paddingRight": 11,
            "paddingTop": 11,
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignSelf": "flex-end",
              "marginBottom": 0,
            },
          ]
        }
        testID="tile-content"
      />
      <Text
        inversed={true}
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "bottom": 8,
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "700",
              "left": 6,
              "marginLeft": 4,
              "position": "absolute",
              "textAlign": "left",
            },
          ]
        }
        testID="tile-title"
        weight="bold"
      >
        Baptism
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onLayout={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": -6,
        "opacity": 1,
        "overflow": "hidden",
        "paddingBottom": 6,
        "paddingLeft": 6,
        "paddingRight": 6,
        "paddingTop": 6,
        "width": "50%",
      }
    }
    testID="tile"
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#1C2530",
            "borderRadius": 5,
            "height": "100%",
            "justifyContent": "space-between",
            "paddingBottom": 11,
            "paddingLeft": 11,
            "paddingRight": 11,
            "paddingTop": 11,
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignSelf": "flex-end",
              "marginBottom": 0,
            },
          ]
        }
        testID="tile-content"
      />
      <Text
        inversed={true}
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "bottom": 8,
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "700",
              "left": 6,
              "marginLeft": 4,
              "position": "absolute",
              "textAlign": "left",
            },
          ]
        }
        testID="tile-title"
        weight="bold"
      >
        Prayer
      </Text>
    </View>
  </View>
</View>
`;
